tests/std/tests/P0896R4_common_iterator_death/test.cpp

```diff
@@ -28,71 +28,158 @@ struct simple_input_iter {
     }
 
-    bool operator==(simple_input_iter const&) const = default;
+    bool operator==(const simple_input_iter&) const = default;
     bool operator==(const default_sentinel_t&) const {
         return true;
     }
 
+    difference_type operator-(const simple_input_iter&) const {
+        return 42;
+    }
+    friend difference_type operator-(const simple_input_iter&, const default_sentinel_t&) {
+        return 42;
+    }
+    friend difference_type operator-(const default_sentinel_t&, const simple_input_iter&) {
+        return 42;
+    }
+
     friend void iter_swap(const simple_input_iter&, const simple_input_iter&) {}
 };
 
 using CIT = common_iterator<simple_input_iter, default_sentinel_t>;
 
-void test_case_operator_dereference() {
+void test_case_operator_dereference_sentinel() {
     CIT cit{default_sentinel};
-    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+    (void) (*cit); // common_iterator can only be dereferenced if it holds an iterator
 }
 
-void test_case_operator_dereference_const() {
+void test_case_operator_dereference_valueless() {
+    CIT cit{_Common_iterator_construct_tag{}};
+    (void) (*cit); // common_iterator can only be dereferenced if it holds an iterator
+}
+
+void test_case_operator_dereference_const_sentinel() {
     const CIT cit{default_sentinel};
-    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+    (void) (*cit); // common_iterator can only be dereferenced if it holds an iterator
+}
+
+void test_case_operator_dereference_const_valueless() {
+    const CIT cit{_Common_iterator_construct_tag{}};
+    (void) (*cit); // common_iterator can only be dereferenced if it holds an iterator
 }
 
-void test_case_operator_arrow() {
+void test_case_operator_arrow_sentinel() {
     CIT cit{default_sentinel};
-    (void) (cit.operator->()); // cannot dereference common_iterator that holds a sentinel
+    (void) (cit.operator->()); // common_iterator can only be dereferenced if it holds an iterator
+}
+void test_case_operator_arrow_valueless() {
+    CIT cit{_Common_iterator_construct_tag{}};
+    (void) (cit.operator->()); // common_iterator can only be dereferenced if it holds an iterator
 }
 
-void test_case_operator_preincrement() {
+void test_case_operator_preincrement_sentinel() {
     CIT cit{default_sentinel};
-    ++cit; // cannot pre increment common_iterator that holds a sentinel
+    ++cit; // common_iterator can only be preincremented if it holds an iterator
 }
 
-void test_case_operator_increment() {
+void test_case_operator_preincrement_valueless() {
+    CIT cit{_Common_iterator_construct_tag{}};
+    ++cit; // common_iterator can only be preincremented if it holds an iterator
+}
+
+void test_case_operator_postincrement_sentinel() {
     CIT cit{default_sentinel};
-    cit++; // cannot increment common_iterator that holds a sentinel
+    cit++; // common_iterator can only be postreincremented if it holds an iterator

```

Comment:
I think you accidentally put the error message for the "postreincrement" operator here. ;)